pandas/core/groupby/generic.py

```diff
@@ -262,10 +262,7 @@ def aggregate(self, func=None, *args, **kwargs):
 
             try:
                 return self._python_agg_general(func, *args, **kwargs)
-            except (ValueError, KeyError, AttributeError, IndexError):
-                # TODO: IndexError can be removed here following GH#29106
-                # TODO: AttributeError is caused by _index_data hijinx in
-                #  libreduction, can be removed after GH#29160
```

Comment:
Why can this AttributeError now be removed? (the mentioned PR wasn't merged) 
It's solved in a different way?

Reply:
This was fixed as a side-effect in #29548.  It turned out `SeriesGrouper` was incorrectly failing to update `_index_data`, only `SeriesBinGrouper` was doing it correctly.

Reply:
Ah, so it was the `_data` -> `_index_data` change there. I see!